src/reducers/entities/posts.js

```diff
@@ -31,6 +31,35 @@ function handleReceivedPost(posts = {}, postsInChannel = {}, action) {
     return {posts: nextPosts, postsInChannel: nextPostsForChannel};
 }
 
+function handleRemovePendingPost(posts = {}, postsInChannel = {}, action) {
+    const postId = action.data.id;
+    const channelId = action.data.channel_id;
+
+    const nextPosts = {
+        ...posts,
+    };
+
+    Reflect.deleteProperty(nextPosts, postId);
+
+    let nextPostsForChannel = postsInChannel;
+
+    // Only change postsInChannel if the order of the posts needs to change
+    if (!postsInChannel[channelId] || postsInChannel[channelId].indexOf(postId) !== -1) {
+        // If we don't already have the post, assume it's the most recent one
+        const postsForChannel = postsInChannel[channelId] || [];
+
+        nextPostsForChannel = {...postsInChannel};
+        nextPostsForChannel[channelId] = [];
+        for (const post of postsForChannel) {
```

Comment:
Is this really necessary? Why not just get the postsInCannel and just remove the post Id if it exists

Reply:
To remove an element from an array in javascript you have to search and remove it anyway. I changed it with a more compact and concrete way to do it, but it is still a loop.